bearer/bearer_test.go

```diff
@@ -214,235 +287,673 @@ func TestToken_InvalidAt(t *testing.T) {
+	require.Zero(t, val.Iat())
+	require.Zero(t, val.Nbf())
+	require.True(t, val.AssertContainer(cidtest.ID()))
+	require.True(t, val.AssertUser(usertest.ID()))
+	_, ok = val.Signature()
+	require.False(t, ok)
+
+	// filled
+	err := val.UnmarshalSignedData(validSignedBearerToken)
 	require.NoError(t, err)
-
-	val2 = bearertest.Token()
-	require.NoError(t, val2.UnmarshalJSON(jd))
-	require.True(t, val2.VerifySignature())
+	val.AttachSignature(anyValidSignature)
+	t.Skip("https://github.com/nspcc-dev/neofs-sdk-go/issues/606")
+	require.Equal(t, validBearerToken, val)
 }
 
-func TestToken_SignedData(t *testing.T) {
+func TestToken_AttachSignature(t *testing.T) {
 	var val bearer.Token
-
-	require.False(t, val.VerifySignature())
-
-	signedData := val.SignedData()
-	var dec bearer.Token
-	require.NoError(t, dec.UnmarshalSignedData(signedData))
-	require.Equal(t, val, dec)
-
-	usr := usertest.User()
-	val = bearertest.Token()
-	val.SetIssuer(usr.UserID())
-
-	usertest.TestSignedData(t, usr, &val)
+	_, ok := val.Signature()
+	require.False(t, ok)
+	val.AttachSignature(anyValidSignature)
+	sig, ok := val.Signature()
+	require.True(t, ok)
+	require.Equal(t, anyValidSignature, sig)
 }
 
 func TestToken_ReadFromV2(t *testing.T) {
-	var val bearer.Token
+	var lt acl.TokenLifetime
+	lt.SetExp(anyValidExp)
+	lt.SetIat(anyValidIat)
+	lt.SetNbf(anyValidNbf)
+	var ms refs.OwnerID
+	ms.SetValue(anyValidSubject[:])
+	var mi refs.OwnerID
+	mi.SetValue(anyValidIssuer[:])
+	var mb acl.BearerTokenBody
+	mb.SetEACL(anyValidEACL.ToV2())
+	mb.SetOwnerID(&ms)
+	mb.SetIssuer(&mi)
+	mb.SetLifetime(&lt)
+	var msig refs.Signature
+	anyValidSignature.WriteToV2(&msig)
 	var m acl.BearerToken
+	m.SetBody(&mb)
+	m.SetSignature(&msig)
 
-	require.Error(t, val.ReadFromV2(m))
-
-	var body acl.BearerTokenBody
-	m.SetBody(&body)
-
-	require.Error(t, val.ReadFromV2(m))
-
-	eaclTable := eacltest.Table()
-	eaclTableV2 := eaclTable.ToV2()
-	body.SetEACL(eaclTableV2)
-
-	require.Error(t, val.ReadFromV2(m))
-
-	var lifetime acl.TokenLifetime
-	body.SetLifetime(&lifetime)
-
-	require.Error(t, val.ReadFromV2(m))
-
-	const iat, nbf, exp = 1, 2, 3
-	lifetime.SetIat(iat)
-	lifetime.SetNbf(nbf)
-	lifetime.SetExp(exp)
-
-	body.SetLifetime(&lifetime)
-
-	require.Error(t, val.ReadFromV2(m))
-
-	var sig refs.Signature
-	m.SetSignature(&sig)
-
-	require.NoError(t, val.ReadFromV2(m))
-
-	var m2 acl.BearerToken
-
-	val.WriteToV2(&m2)
-	require.Equal(t, m, m2)
-
-	usr, usr2 := usertest.ID(), usertest.ID()
-
-	require.True(t, val.AssertUser(usr))
-	require.True(t, val.AssertUser(usr2))
-
-	var usrV2 refs.OwnerID
-	usr.WriteToV2(&usrV2)
-
-	body.SetOwnerID(&usrV2)
-
-	require.NoError(t, val.ReadFromV2(m))
-
-	val.WriteToV2(&m2)
-	require.Equal(t, m, m2)
-
-	require.True(t, val.AssertUser(usr))
-	require.False(t, val.AssertUser(usr2))
-
-	var s neofscrypto.Signature
-
-	require.NoError(t, s.CalculateMarshalled(neofscryptotest.Signer(), &body, nil))
-
-	s.WriteToV2(&sig)
-
+	var val bearer.Token
```

Comment:
Could you please reorganise this code? Add some blank lines for instance. It is difficult to read it.

The same for some other places

Reply:
added \n